### ISSUE TITLE
Carry enqueued videos over if accidentally click new video

### ISSUE THREAD

**Issue by mecuricat** [TeamNewPipe/NewPipe#10099]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

If you click a new video, prompt user whether to keep/carry over enqueued videos if available.

### Why do you want this feature?

Let's say, you are watching a video while 7 other videos are enqueued. Then you accidentally touch a new video and all your enqueued videos history is lost. You don't remember them all.

### Additional information

Some users may not like to see this prompt window asking them about enqueue history. So, perhaps a tick box can be provided additionally to stop showing the window again.

**Comment by mecuricat** [comment]
> What does the "Ask for confirmation before clearing queue" toggle in Video and audio settings do?

The description of this setting says that "Switching from one player to another may replace your queue".

But I am not switching players. I am actively watching a video in the main player while 7 videos are enqueued. While the video is playing, I am browsing through other videos. Then an accidental touch on a different video thumbnail starts that video forgetting my queue history.

**Comment by CouldBeThis** [comment]
I would just like to say that I have this issue also and it drives me nuts. In contrast to my otherwise amazing experience with this app.  I am constantly deleting my whole queue. Sometimes because of touching another video like @mecuricat mentioned. Other times, I don't even know what I do to provoke it. The other day I did it 3-4 times by picking up my phone with an unlocked screen. 

Personally, I would prefer to *never* automatically clear the queue. If a new video will start playing immediately, then add it to the top of the queue and continue on after that. It is easy enough to clear the queue manually when needed. Well I assume it is anyway. Don't think I have ever had one around long enough to require the operation as they are so fragile. :)  

Maybe a setting along the lines of 

Action to take when video(s) already playing and another video is opened:
 🔾 Delete the existing queue, and start playing the new video 
 🔾 Add the new video to the top of the queue, and start playing it immediately
 🔾 Add the new video next in the queue, to be played after the current video is done
 🔾 Add the new video to the bottom of the queue 
 🔾 Do not take any action

Hopefully someone will come up with a more concise description. That's all I am able to contribute though. Thanks for the great app!

**Comment by nerdguy1338** [comment]
This is my only complaint so far with this app.

In this instance, the default behavior should be append to queue.

**Comment by tutacat** [comment]
> @CouldBeThis Maybe a setting along the lines of
>
> Action to take when video(s) already playing and another video is opened:
> 🔾 Delete the existing queue, and start playing the new video
> 🔾 Add the new video to the top of the queue, and start playing it immediately
> 🔾 Add the new video next in the queue, to be played after the current video is done
> 🔾 Add the new video to the bottom of the queue
> 🔾 Do not take any action

This seems to be a bit too complex. It think it's easier to follow similar function as in other existing programs. I think it would make sense to be:
- When a queue/video is open, manually close/delete the queue to remove the queue. Maybe have an option to replace in the context / more options menu.
- When queue/video is playing, and video is clicked, insert it after the current video in queue. Then pause the current content, switch to the new content's info page (in the queue), and start playing if auto-play is enabled. If "ask before replacing queue" or replacement setting is enabled, ask before pausing and switching to the new content page.

We already have the queue adding options (end of queue, and next in queue) with the context menu.

**Comment by mimmo9669** [comment]
+1

**Comment by nerdguy1338** [comment]
Bumping this. It's my only issue with this app. Even the YouTube casting queue from my phone to various apps is less fragile. I can switch casting locations just fine, and my queue stays 

Why is this seemingly so hard to fix? It's been  almost a year.

**Comment by tutacat** [comment]
> Why is this seemingly so hard to fix? It's been almost a year.

Because it's actually a feature request, plus it takes a community project more time to implement a feature or start implementing a feature than accepting any pulls.

It has actually gotten better because they have changed how directly the player view is linked to the queue, but of course they are refactoring everything to make it work better. That will definitely take priority. Currently it stores the last played single video, but not the queue.

I will look at hacking this because it should be stored to disk to help continuity from crashes / reboots, though not sure what they will be looking at integrating just yet, due to the refactor.

Interestingly, in the official app (not website) it kind of a paid feature to use a queue properly -- it has history feature but doesn't let you use the queue feature properly.

**Comment by nerdguy1338** [comment]
How about this? When the queue is made, meaning more than 0 videos in it,
make it a playlist in the "bookmarked playlist" menu.

**Comment by tutacat** [comment]
@nerdguy1338 

Yes this feels like a useful feature. I think it should be a stack so you have the latest queue, and a history of queues up to a limit, but can quickly be moved to a separate playlist.

**Comment by nerdguy1338** [comment]
Make the limit user- configurable. Start at 10000.

**Comment by tutacat** [comment]
I was thinking time based is better

**Comment by GfEW** [comment]
@nerdguy1338 
> How about this? When the queue is made, meaning more than 0 videos in it, make it a playlist in the "bookmarked playlist" menu.

Not everybody wants the list of playlists to get cluttered by huge numbers of auto-saved queues, but making the *current* queue the default (anonymous) playlist that
* is automatically preserved and restored when applicable (e. g. after crashes),
* is accessible and editable like any other playlist, and 
* can be given a custom name to become a "real playlist",

would be a most welcome improvement.

**Comment by nerdguy1338** [comment]
> @tutacat
> 
> > @nerdguy1338
> > Yes this feels like a useful feature. I think it should be a stack so you have the latest queue, and a history of queues up to a limit, but can quickly be moved to a separate playlist.
> 
> Not everybody wants the list of playlists to get cluttered by huge numbers of auto-saved queues, but making the _current_ queue the default (anonymous) playlist that
> 
> * can be populated from the context menu in video lists, or by long-pressing the background/popup icons in details view
>   (already implemented),
> * is automatically preserved and restored when applicable, e. g. after crashes
>   (currently lacking),
> * is accessible and editable from the playlists listing like any other playlist
>   (partially implemented), and
> * can be given a custom name to become a "real" playlist
>   (currently lacking),
> 
> would be a most welcome improvement.

While we're tweaking things, the ability to merge bookmarked (or custom newpipe-only) playlists into on bigger one and remove duplicates, would be nice.

**Comment by emutier** [comment]
Please count each click as "Enqueue" once a queue exists. This is how youtube (premium) solved it and I think it is the most intuitive way to do.
